Sources/MapboxNavigation/UIImage.swift

```diff
@@ -15,22 +15,35 @@ extension UIImage {
         self.init(cgImage: cgImage)
     }
     
-    func withCenteredText(_ text: String, color: UIColor, font: UIFont, scale: CGFloat) -> UIImage {
-        let maxHeight = font.lineHeight * 2
-        var constrainedSize = size
-        constrainedSize.width = min(constrainedSize.width, constrainedSize.width * maxHeight / constrainedSize.height)
-        constrainedSize.height = min(constrainedSize.height, maxHeight)
+    func withCenteredText(_ text: String,
+                          color: UIColor,
+                          font: UIFont,
+                          size: CGSize? = nil) -> UIImage {
+        let maxHeight = size?.height ?? font.lineHeight * 2
```

Comment:
In case if size is set we'll use it to provide maximum height for the image. In this case this height is height of `UILabel` container.